### PR TITLE
Frontend app

### PR DESCRIPTION
**Pull request by dn1470**
Typescript Github Oauth

### CODE REVIEW

File: .vscode/dryrun.log [hunk 1]
```diff
@@ -0,0 +1,5 @@
+make --dry-run --always-make --keep-going --print-directory
+make: Entering directory '/home/youngju/comp9447-team2'
+CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/main cmd/*.go
+make: Leaving directory '/home/youngju/comp9447-team2'
```
Comment (The-TT-Hacker) [inline .vscode/dryrun.log]: again dont add files with your personal stuff in it

File: frontend/package.json [hunk 2]
```diff
@@ -47,16 +48,9 @@
     "@types/node": "^18.0.4",
     "@types/react": "^18.0.15",
     "@types/react-dom": "^18.0.6",
-    "@typescript-eslint/eslint-plugin": "^5.30.6",
-    "@typescript-eslint/parser": "^5.30.6",
-    "eslint-config-airbnb": "^19.0.4",
-    "eslint-config-airbnb-typescript": "^17.0.0",
-    "eslint-config-prettier": "^8.5.0",
-    "eslint-plugin-jest": "^26.6.0",
-    "eslint-plugin-prettier": "^4.2.1",
```
Comment (The-TT-Hacker) [inline frontend/package.json:56]: why did you remove all of these?

File: frontend/package.json [hunk 3]
```diff
@@ -47,16 +48,9 @@
...
     "openapi-typescript-codegen": "^0.23.0",
     "prettier": "^2.7.1",
-    "react-scripts": "^5.0.1",
+    "react-scripts": "^2.1.3",
```
Comment (The-TT-Hacker) [inline frontend/package.json:53]: why did you change back to a version thats 4 years old?

File: frontend/src/components/Utils/Avatar.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+const Avatar = (props: AvatarProps) => {
+  return (
+    <div>
+      <>
+        <h2>{props.name}</h2>
+        <img className="avatar" src={props.avatar} alt="user image" />
+        <a className="logout"
+          href={`http://localhost:5000/logout`}
```
Comment (The-TT-Hacker) [inline frontend/src/components/Utils/Avatar.tsx]: use the backend variable here

File: frontend/src/pages/login.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React, { useEffect, useState } from 'react';
+
+const Login = () => {
+  const GITHUB_CLIENT_ID = 'c2f25de6e7cfc7713944';
+  const gitHubRedirectURL = 'http://localhost:8080/api/auth/github';
```
Comment (The-TT-Hacker) [inline frontend/src/pages/login.tsx:5]: this stuff is in your other PR as well. whats going on with that one?

**Comment by The-TT-Hacker** [comment]
filenames for components should be in capital case. e.g. GithubLogin.tsx

File: terraform/aws_only/ecr.tf [hunk 6]
```diff
@@ -15,4 +15,4 @@ resource "aws_ecr_repository" "image_repository" {
     encryption_type = "KMS"
     kms_key         = aws_kms_key.ecr_kms.key_id
   }
-}
```
Comment (The-TT-Hacker) [inline terraform/aws_only/ecr.tf:18]: dont add stuff to unrelated files

File: frontend/tsconfig.json [hunk 7]
```diff
@@ -19,7 +19,7 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react-jsx"
+    "jsx": "preserve"
```
Comment (The-TT-Hacker) [inline frontend/tsconfig.json:22]: whats the reason for this?

File: frontend/src/pages/userpage.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  }, [])
+
+  
+  
+  return (
+    <div className="center">
+      {datafetched === false ? <h1>Loading User Data...</h1> :
+        <>
+        <p>{userData}</p>
+        {/* <Avatar avatar={userData.avatar_url} name={userData.name} />
```
Comment (The-TT-Hacker) [inline frontend/src/pages/userpage.tsx]: dont leave in commented out code

File: frontend/src/pages/userpage.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        })
+
+  }
+
+
+  useEffect(() => {
+    fetch('/api')
+        .then(res => res.json())
+        .then((data) => {
+            console.log(data.username)
```
Comment (The-TT-Hacker) [inline frontend/src/pages/userpage.tsx:31]: dont leave in console.logs

File: frontend/src/pages/userpage.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+  const navigate = useNavigate();
+
+  function fetchUserData(name: React.ElementType<any>) {
+
+    fetch(`https://api.github.com/users/${name}`)
+        .then(res => res.json())
+        .then(data => {
+            setUserData(data)
+            console.log(data)
```
Comment (The-TT-Hacker) [inline frontend/src/pages/userpage.tsx:21]: dont leave in console.logs

File: frontend/src/pages/usepage.tsx [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+            console.log(data)
+        })
+
+  }
+
+  useEffect(() => {
+    fetch('/api')
+        .then(res => res.json())
+        .then((data) => {
+            // console.log(data.username)
```
Comment (The-TT-Hacker) [inline frontend/src/pages/usepage.tsx:30]: remove commented out code

File: frontend/package.json (hunk 2 above)
Comment (dn1470) [reply frontend/package.json:56]: I was following npm instruction. It said new version of eslint is required to install @type/react but there were some sort of conflicts and it told me delete package.lock file and remove eslint lines from package.json

Comment (The-TT-Hacker) [reply frontend/package.json:56]: i dont see a reason to touch any of this stuff or the react versions. it is all working fine

Comment (dn1470) [reply frontend/package.json:56]: I don't know what exactly causing the issue and I never used typescript before but importing react and react router dom didn't work with the original setting.

Comment (The-TT-Hacker) [reply frontend/package.json:56]: it worked fine in Rubaiats PR
